src/lib.rs

```diff
@@ -536,7 +546,7 @@ impl<B: BitBlock> BitVec<B> {
     #[inline]
     pub fn set(&mut self, i: usize, x: bool) {
         self.ensure_invariant();
-        assert!(i < self.nbits, "index out of bounds: {:?} >= {:?}", i, self.nbits);
+        debug_assert!(i < self.nbits, "index out of bounds: {:?} >= {:?}", i, self.nbits);
```

Comment:
If self.nbits is only part of a storage unit, this will incorrectly allow it through.

Example:

```rust
storage: [0b_0000_1010_u8],
nbits: 4,
```

This will allow `bitvec.set(6, whatever)`, even though it is outside the current range.